### ISSUE TITLE
Disable unused layout templates

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#5068]
**Description of the need**

I've just implemented `hook_theme_registry_alter()` in a custom block and checked the registry content to see long list of variables, including for unused layout templates. Disabling them on `admin/structure/layouts/settings` freed registry from unnecessary content and even-though human eyes might not notice, definitely made things slightly faster.

**Proposed solution**

I believe Backdrop must be shipped as performant as possible, so I suggest to disable all unused layout templates by default. Users can easily enable them if they need additional templates. 

Additionally, we could add a pointer link on pages like `admin/structure/layouts/manage/dashboard/configure` as shown below:

![Screen Shot 2021-04-24 at 10 24 26 PM](https://user-images.githubusercontent.com/56604405/115980704-f8c93c00-a54b-11eb-897c-4a1afa22716b.png)

**Comment by klonos** [comment]
I'm afraid I don't agree with this proposal 👎🏼 

- I would like to see hard data re performance gain. Is it enough to warrant any first-time out-of-the-box UX regression?
- It's easier for people to disable/tweak things than to discover/enable them, and we are making assumptions here re people reading help text (most people don't). If we were to go ahead with something like this, then instead of help text I would like to see a pseudo-tile, with a short text like "Enable more templates":
  <img width="50%" height="50%" src="https://user-images.githubusercontent.com/2423362/115982647-71a8b200-a5a5-11eb-82bf-aa2883678377.png">
- This seems to me similar to the "disable Views UI when not needed" best practice situation: I find it to be a post-build task, after a site has been launched - not an OOTB optimization.

**Comment by alanmels** [comment]
> I find it to be a post-build task

Fair enough, this was just a suggestion. At the same time, you sound from completely developer's point, which is natural as you are one indeed. However, while it's easy for developers to bend Backdrop whatever way they want at any stage or step, not everything obvious for the regular users. And I'm afraid the most Backdrop sites in the wild will end up with all the layout templates enabled without users realizing they can just disable to make little bit less overhead.

**Comment by klonos** [comment]
> ...you sound from completely developer's point, which is natural as you are one indeed.

Well, not a developer, rather than a (no-code/low-code) site builder. But yes, I always try to see the UX from that point of view (as less coding, and as much being possible via the admin UI). With Drupal I felt that this was being neglected in favor of things like composer/symfony/docker etc. That frustrated me and still annoys me to this day, even though I know how to use all these "develop-y" things now.
